src/inference/smc.js

```diff
@@ -145,7 +159,10 @@ module.exports = function(env) {
     var logW = util.logsumexp(_.map(particles, 'logWeight'));
     var logAvgW = logW - Math.log(m);
 
-    assert.notStrictEqual(logAvgW, -Infinity, 'All particles have zero weight.');
+    if (logAvgW === -Infinity) {
+      // do not return, execution continues
+      env.coroutine.error('All particles have zero weight.');
+    }
```

Comment:
I agree that this assertion should be an error, but I believe the remaining assertions in SMC are better left unchanged. (See my original comment for an attempt to explain my thinking.) Are you OK to revert these other ones?